### DISCUSSION TITLE
Is it possible to convert classical Pytorch networks to SNN?

### DISCUSSION THREAD

**Discussion by JNaranjo-Alcazar** [lava-nc/lava#465] [Q&A] [answered]
Hi,

From what I see in the examples, networks must be defined with specific library layers such as Boostrap in order to be converted to SNN. 

My question is the following:
if I have a previously trained network, say a CNN (defined with Pytorch layers) and I have it saved in a .pt file, can I convert it to SNN using the LAVA library?

Best regards and thanks for your time

**Comment by bamsumit** [comment, marked as answer]
Hi @JNaranjo-Alcazar, currently there is no support for ANN-SNN conversion in lava-dl. Bootstrap in lava-dl is ANN-SNN co-training and they need to trained together, and thus do not work with pretrained networks.

There are other ANN-SNN conversion frameworks like [SNN Toolbox](https://github.com/NeuromorphicProcessorProject/snn_toolbox) (although it's tensorflow based). One thing you should be aware is that _typical ANN-SNN conversion networks tend to need a large number of timesteps, therefore, are not so efficient on neuromorphic hardware_.

In any case, you can write your own script to describe the translated SNN network in hdf5 format (description [here](https://github.com/lava-nc/lava-dl/blob/main/src/lava/lib/dl/netx/README.md#hdf5-description-protocol)) and run inference on it with lava-dl NetX. I am happy to help answer any queries on the protocol.
